tests/queries/0_stateless/02457_datediff_via_unix_epoch.sql

```diff
@@ -0,0 +1,23 @@
+select 'year', date_diff('year', toDate32('1969-12-25'), toDate32('1970-01-05'));
+select 'year', date_diff('year', toDateTime64('1969-12-25 10:00:00.000', 3), toDateTime64('1970-01-05 10:00:00.000', 3));
+
+select 'quarter', date_diff('quarter', toDate32('1969-12-25'), toDate32('1970-01-05'));
+select 'quarter', date_diff('quarter', toDateTime64('1969-12-25 10:00:00.000', 3), toDateTime64('1970-01-05 10:00:00.000', 3));
+
+select 'month', date_diff('month', toDate32('1969-12-25'), toDate32('1970-01-05'));
+select 'month', date_diff('month', toDateTime64('1969-12-25 10:00:00.000', 3), toDateTime64('1970-01-05 10:00:00.000', 3));
+
+select 'week', date_diff('week', toDate32('1969-12-25'), toDate32('1970-01-05'));
+select 'week', date_diff('week', toDateTime64('1969-12-25 10:00:00.000', 3), toDateTime64('1970-01-05 10:00:00.000', 3));
+
+select 'day', date_diff('day', toDate32('1969-12-25'), toDate32('1970-01-05'));
+select 'day', date_diff('day', toDateTime64('1969-12-25 10:00:00.000', 3), toDateTime64('1970-01-05 10:00:00.000', 3));
+
+select 'hour', date_diff('hour', toDate32('1969-12-25'), toDate32('1970-01-05'));
+select 'hour', date_diff('hour', toDateTime64('1969-12-25 10:00:00.000', 3), toDateTime64('1970-01-05 10:00:00.000', 3));
+
+select 'minute', date_diff('minute', toDate32('1969-12-31'), toDate32('1970-01-01'));
+select 'minute', date_diff('minute', toDateTime64('1969-12-31 23:50:00.000', 3), toDateTime64('1970-01-01 00:10:00.000', 3));
+
+select 'second', date_diff('second', toDate32('1969-12-31'), toDate32('1970-01-01'));
+select 'second', date_diff('second', toDateTime64('1969-12-31 23:50:00.000', 3), toDateTime64('1970-01-01 00:10:00.000', 3));
```

Comment:
Since `toStableRelativeHourNum` is called for possible argument types (not only `Date32`), it would surely be good to add tests for Date/DateTime/DateTime64 arguments < and > 1970 as well. But totally fine to do this in a separate PR.